lib/helper/TestCafe.js

```diff
@@ -135,37 +151,60 @@ class TestCafe extends Helper {
 
     this.iteration += 2; // Use different ports for each test run
     // @ts-ignore
-    this.testcafe = await createTestCafe('localhost', null, null);
+    this.testcafe = await createTestCafe('', null, null);
 
     this.debugSection('_before', 'Starting testcafe browser...');
 
     this.isRunning = true;
 
     // TODO Do we have to cleanup the runner?
     const runner = this.testcafe.createRunner();
-    runner
-      .src(this.dummyTestcafeFile)
-      .screenshots(global.output_dir, !this.options.disableScreenshots)
-      // .video(global.output_dir) // TODO Make this configurable
-      .browsers(this.options.show ? this.options.browser : `${this.options.browser}:headless`)
-      .reporter('minimal')
-      .run({
-        skipJsErrors: true,
-        skipUncaughtErrors: true,
-        quarantineMode: false,
-        // debugMode: true,
-        // debugOnFail: true,
-        // developmentMode: true,
-        pageLoadTimeout: this.options.getPageTimeout,
-        selectorTimeout: this.options.waitForTimeout,
-        assertionTimeout: this.options.waitForTimeout,
-        takeScreenshotsOnFails: true,
-      })
-      .catch((err) => {
-        this.debugSection('_before', `Error ${err.toString()}`);
-        this.isRunning = false;
-        this.testcafe.close();
+
+    if (this.options.browser !== 'remote') {
```

Comment:
Could you split this long if/else into 2 functions?

* `_startRemoteBrowser`
* `_startBrowser`

`_` is needed so the function to be private and not available from a test.

Reply:
I agree that is an improvement. I have implemented the changes.